lib/ui/views/home/home_viewmodel.dart

```diff
@@ -492,8 +492,8 @@ class HomeViewModel extends BaseViewModel {
   }
 
   Future<void> forceRefresh(BuildContext context) async {
-    _managerAPI.clearAllData();
+    await _managerAPI.clearAllData();
+    await initialize(context);
```

Comment:
Whats the reason this line is changed?

Reply:
That's a good question, and the answer is quite simple. 

As @ILoveOpenSourceApplications pointed out, the refresh animation appears to cancel because it automatically resolves after being started. 

The `RefreshIndicator` plays the refreshing animation until the `onRefresh Future<void>` resolves. Since the method we use in the code, `forceRefresh`, doesn't properly await its asynchronous callbacks, the function immediately stops/resolves milliseconds after being called. It is for this reason that I changed the function to:

1. Wait for the inner promises to resolve 
2. Changed the order to prevent the "refresh done" before actually doing the work 
3. Complete the promise, stopping the animation properly with patience (waiting) instead of rushing (no-waiting)

> Flutter Docs 
> [onRefresh](https://api.flutter.dev/flutter/material/RefreshIndicator/onRefresh.html) → [RefreshCallback](https://api.flutter.dev/flutter/material/RefreshCallback.html)
>
> A function that's called when the user has dragged the refresh indicator far enough to demonstrate that they want the app to refresh. The returned [Future](https://api.flutter.dev/flutter/dart-async/Future-class.html) must complete when the refresh operation is finished.